engine/src/main/java/org/destinationsol/menu/BackgroundAsteroidManager.java

```diff
@@ -0,0 +1,116 @@
+
+            float distance = (float)Math.sqrt(Math.pow(BackgroundAsteroid.getPosition().x - displayDimensions.getRatio()/2, 2) + Math.pow(BackgroundAsteroid.getPosition().y - 0.5f, 2));
+            if (distance < radius) {
+                retainedBackgroundAsteroids.add(BackgroundAsteroid);
+            } else {
+                retainedBackgroundAsteroids.add(buildAsteroid());
+            }
+        }
+
+        BackgroundAsteroids.clear();
+        BackgroundAsteroids.addAll(retainedBackgroundAsteroids);
+    }
+
+    public void draw(UiDrawer uiDrawer) {
+        for (BackgroundAsteroid BackgroundAsteroid : BackgroundAsteroids) {
+            BackgroundAsteroid.draw(uiDrawer);
+        }
+    }
+
+    public BackgroundAsteroid buildAsteroid() {
+
+        TextureAtlas.AtlasRegion texture = Assets.getAtlasRegion(SolRandom.test(0.5f) ? "engine:asteroid_0" : "engine:asteroid_1");
+
+        boolean small = SolRandom.test(.8f);
+        float scale = (small ? .1f : .4f) * SolRandom.randomFloat(.5f, 1);
+        Color tint = new Color();
+        SolColorUtil.fromHSB(SolRandom.randomFloat(0, 1), .25f, 1, .7f, tint);
+
+        float radiusX = (float) (texture.originalHeight) / displayDimensions.getWidth() * scale / 2;
+        float radiusY = (float) (texture.originalHeight) / displayDimensions.getHeight() * scale / 2;
+
+        float r = displayDimensions.getRatio();
+        Vector2 velocity, position;
+        if (SolRandom.test(0.5f)) {
+            // Spawn to the left or right of screen
+            boolean toLeft = SolRandom.test(1f);
+            velocity = new Vector2((float) Math.pow(SolRandom.randomFloat(toLeft ? 0.025f : -0.1f, toLeft ? 0.1f : 0.025f), 2), (float) Math.pow(SolRandom.randomFloat(0.095f), 2));
+            position = new Vector2(r / 2 + (toLeft ? -1 : 1) * (r / 2 + radiusX) - radiusX, 0.5f + SolRandom.randomFloat(0.5f + radiusY) - radiusY);
+        } else {
+            // Spawn at the top or bottom of screen
+            boolean atTop = SolRandom.test(1f);
+            velocity = new Vector2((float) Math.pow(SolRandom.randomFloat(0.095f), 3), (float) Math.pow(SolRandom.randomFloat(atTop ? -0.025f : 0.025f, atTop ? -0.1f : 0.1f), 2));
+            position = new Vector2(r / 2 + SolRandom.randomFloat(r / 2 + radiusX) - radiusX, 0.5f + (atTop ? -1 : 1) * (0.5f + radiusY) - radiusY);
+        }
+
+        float angle = SolRandom.randomFloat((float) Math.PI);
+        float angularVelocity = SolRandom.randomFloat(1.5f);
+        velocity.scl(50);
+
+        Body body = asteroidMeshLoader.getBodyAndSprite(texture, scale*0.905f, BodyDef.BodyType.DynamicBody, position, angle, new ArrayList<>(), 10f, DrawableLevel.BODIES);
+        body.setLinearVelocity(velocity);
+        body.setAngularVelocity(angularVelocity);
+        BackgroundAsteroid asteroid = new BackgroundAsteroid(texture, scale, tint, position, velocity, angle, body);
+        body.setUserData(asteroid);
+
+        return asteroid;
+    }
+
```

Comment:
You could probably decompose this function into smaller functions. makes it a bit easier to understand that way. Also, unnecessary space here.